azure/scope/machine.go

```diff
@@ -294,10 +294,9 @@ func (m *MachineScope) BuildNICSpec(nicName string, infrav1NetworkInterface infr
 		if m.Role() == infrav1.ControlPlane {
 			spec.PublicLBName = m.OutboundLBName(m.Role())
 			spec.PublicLBAddressPoolName = m.OutboundPoolName(m.Role())
-			if m.IsAPIServerPrivate() {
-				spec.InternalLBName = m.APIServerLBName()
-				spec.InternalLBAddressPoolName = m.APIServerLBPoolName()
-			} else {
+			spec.InternalLBName = m.APIServerLBName() + "-ilb"
+			spec.InternalLBAddressPoolName = m.APIServerLBPoolName() + "-ilb"
+			if !m.IsAPIServerPrivate() {
```

Comment:
```suggestion
			if m.IsAPIServerPrivate() {
				spec.InternalLBName = m.APIServerLBName()
				spec.InternalLBAddressPoolName = m.APIServerLBPoolName()
			} else {
				spec.InternalLBName = m.APIServerLBName() + "-ilb"
				spec.InternalLBAddressPoolName = m.APIServerLBPoolName() + "-ilb"
				spec.PublicLBNATRuleName = m.Name()
				spec.PublicLBAddressPoolName = m.APIServerLBPoolName()
			}
		}
```

I think this needs to be updated as above.

Reply:
Ignore commit name, but addressed this in the commit 0b5d19c017c66e19a5dd9950a3c4adfd031d2d9a